.github/coding-style.md

```diff
@@ -0,0 +1,171 @@
+- Avoid `var ( … )` blocks within function scopes, only use them at the global
+  scope.
+- Declarations at the global scope should follow this order: const block, var
+  block, everything else. If you define struct types and methods of this type in
+  the same file, the order should be: type declaration, new/constructor function,
+  methods. Do not interleave methods of different types.
+- If you are registering packages via `_` then extract this line into its own
+  import block and preface it with a comment explaining what this import does.
+- Scope variables as low as possible.
+- Only use `if` + assignment with `:=`.
+
+### Functions
+
+- Every function that can potentially block should receive a `ctx context.Context`
+  as the first parameter.
+- An exported function should only ever return exported types (even though Go
+  permits otherwise).
+- Always pass around proto objects as pointers when traversing function
+  boundaries. If you need to make a copy, use `obj.Clone()`.
+- When using slice tricks like filtering in a function, pass the slice as a
+  pointer to explicitly call out that the underlying data may be modified in
+  that function.
+- Avoid naked returns
+
+### Types and collections
+
+- Use `.GetField()` instead of `.Field` on protobuf objects
+- Do not embed public types that are part of an object’s *internal* state into
+  structs, exported or non-exported. I.e., do `mutex sync.Mutex` instead of
+  `sync.Mutex` at the `struct` level. `obj.Lock()` suggests that Lock would be
+  part of the public interface, which it should not be.
+- When using `append`, preallocate whenever possible.
+- When setting elements in a slice, instantiate like `s := make([]string, 0, size)`
+  and use `append` because this greatly decreases potential index out of bounds
+  errors.
+
+### Error handling
+ 
+- Use `errors.Wrap[f]()` from `github.com/pkg/errors` to add a message when
+  forwarding the error.
+- Use `RoxError.CausedBy[f]()` from `pkg/errox` to add context to an existing
+  message. 
+- Prefer `RoxError.New[f]()` from `pkg/errox` over `errors.Errorf()` from
+  `github.com/pkg/errors` and `errors.New()` from the _builtin_ errors package
+  to assign the error one of the standard classes.
+- Define designated error conditions in the package as global variables.
+- When calling a function that returns an error, always check for `err != nil`
+  before doing anything else with the results.
+- When panic’ing, always use `error` objects as the argument.
+- When doing a type conversion, use the form with two values on the left side of
+  the assignment operator, even if you don’t use the value, e.g.,
+  `x, _ := val.(T)`. The single-valued form may panic. If you know that the type
+  conversion will always succeed, add a comment explaining why.
+
+### Comments and text
+
+- Use `TODO(ROX-XYZ)` for tracking what should be done in a follow-up.
+- Aim for 120 columns max line length; consider 80 columns for comments and text
```

Comment:
I know it's your personal thing to have the 80 max line length, however for the guidelines I would only see 120 max line length as applicable. I'm not entirely sure what the value of the 80 max line length would be over the 120.

Reply:
Readability. There are reasons why every managzine and journal use columns instead of lines going the entire width. I never understood why coders denied centuries of experience : ).

Reply:
Only because sometimes the right part is noisy, perhaps, I'm not sure:
```
doSomething()
obj = SimpleObjectNameExampleFactorySingleton().CreateSimpleObjectNameExample(blabla, param, pam, pam, tro, lo, lo)
doSomethingElse(obj)
```
vs
```
doSomething()
obj = SimpleObjectNameExampleFactorySingleton().
  CreateSimpleObjectNameExample(
    blabla, param, pam, pam, tro, lo, lo)
doSomethingElse(obj)
```

Reply:
Yeah, that's why it says "aim" and "consider". It's ok to deviate, as long as the linter is happy.